crates/parser/src/grammar/expressions/atom.rs

```diff
@@ -74,6 +74,14 @@ pub(super) fn atom_expr(p: &mut Parser, r: Restrictions) -> Option<(CompletedMar
         T![|] => closure_expr(p),
         T![move] if la == T![|] => closure_expr(p),
         T![async] if la == T![|] || (la == T![move] && p.nth(2) == T![|]) => closure_expr(p),
+        T![static]
+            if la == T![|]
+                || (la == T![move] && p.nth(2) == T![|])
+                || (la == T![async] && p.nth(2) == T![|])
+                || (la == T![async] && p.nth(2) == T![move] && p.nth(3) == T![|]) =>
+        {
+            closure_expr(p)
+        }
```

Comment:
```suggestion
        (T![async] | T![move] | T![static]) if la == T![|] => closure_expr(p),
        (T![async] | T![static]) if la == T![move] && p.nth(2) == T![|] => closure_expr(p),
        T![static] if la == T![async] && p.nth(2) == T![|] => closure_expr(p),
        T![static] if la == T![async] && p.nth(2) == T![move] && p.nth(3) == T![|]) => closure_expr(p),
```
This can probably be arranged a bit nicer by "number of keywords" prefixing the closure expression.
Similar for the assert below.